R/scanGenome.R

```diff
@@ -727,6 +727,131 @@ getDifferentiallyModifiedWindows <- function(se,
+#'
+#' @returns The \code{\link[GenomicRanges]{GRanges}} object constructed from
+#'     the \code{\link[edgeR]{topTags}} or \code{\link[limma]{topTable}} output
+#'     obtained for the statistical analysis, with an additional summary column
+#'     named "dirNegLog10PValue" (the sign of the logFC multiplied with the
+#'     -log10(PValue)).
+#'
+#' @examples
+#' library(GenomicRanges)
+#' library(SummarizedExperiment)
+#' modbamfiles <- system.file("extdata",
+#'                            c("6mA_1_10reads.bam", "6mA_1_10reads.bam",
+#'                              "6mA_2_10reads.bam", "6mA_2_10reads.bam"),
+#'                            package = "footprintR")
+#' rng <- GRanges("chr1", IRanges(6940000, 6943600))
+#' windowStep <- 180
+#' windowSize <- 4 * windowStep
+#' s <- seq(start(rng), end(rng) - windowSize + 1, by = windowStep)
+#' windowgr <- GRanges(seqnames = seqnames(rng),
+#'                     ranges = IRanges(start = s, width = windowSize))
+#' se <- readModBam(bamfiles = modbamfiles, regions = rng, level = "summary",
+#'                  modbase = "a", trim = TRUE,
+#'                  BPPARAM = BiocParallel::SerialParam())
+#' seFourier <- phasingScoreFourier(se = se, gr = windowgr, numCoef = 5)
+#' seFourier$group <- c("group1", "group1", "group2", "group2")
+#' mm <- model.matrix(~ group, data = colData(seFourier))
+#' gr <- getDifferentialWindows(seFourier, assayName = "phasingScoreAbs",
+#'                              designMatrix = mm,
+#'                              contrast = c(0, 1),
+#'                              method = "limma",
+#'                              verbose = TRUE)
+#' class(gr)
+#' head(gr)
+#'
+#' @importFrom SummarizedExperiment assayNames assay rowRanges
+#' @importFrom GenomicRanges GRanges
+#' @importFrom S4Vectors mcols<- DataFrame
+#' @importFrom methods as
+#' @importFrom cli cli_abort
+#'
+#' @export
+getDifferentialWindows <- function(se,
+                                   assayName = "phasingScoreAbs",
+                                   designMatrix,
+                                   contrast,
+                                   method = "limma",
+                                   verbose = FALSE) {
+    # check arguments
+    .assertVector(x = se, type = "SummarizedExperiment")
+    .assertScalar(x = assayName, type = "character",
+                  validValues = assayNames(se))
+    .assertVector(x = designMatrix, type = "matrix")
+    .assertVector(x = rownames(designMatrix), type = "character")
+    stopifnot(all(rownames(designMatrix) == colnames(se)))
+    .assertVector(x = contrast, type = "numeric", len = ncol(designMatrix))
+    .assertScalar(x = verbose, type = "logical")
+    .assertScalar(x = method, type = "character",
+                  validValues = c("edgeR", "limma"))
+    if (method == "edgeR") {
+        .assertPackagesAvailable(pkgs = "edgeR")
+    } else if (method == "limma") {
+        .assertPackagesAvailable(pkgs = "limma")
+    }
```

Comment:
I guess this could also be just:
`.assertPackagesAvailable(pkgs = method)`

Reply:
Oh yes, good catch 🙂 Fixed